flipperdevices/flipperzero-firmware
# BadUSB: Keyboard Layouts

This is a working and actual version of previous PR #1525 
Originally made by @dummy-decoy and fixed by me

# What's new

- Ability to select keyboard layout in BadUSB app, so `AZERTY`, `QWERTZ` and other ones will work without modifying the badusb scripts
- Tool that can be used to create `.kl` files - https://github.com/dummy-decoy/flipperzero_badusb_kl
- Layouts can be added to microsd card into `badusb/assets/layouts/` folder

# Verification 

- For this example your pc should be configured to use dvorak layout (or any other one non QWERTY)
- Open demo script in badusb app, click left button to open config menu, select Keyboard layout
- Select your layout, for example dvorak
- Run script - verify that it works correct
- Select regular ANSI QWERTY layout en-US for example while your PC has dvorak layout
- Try again, see that script works same as on firmware without these change

# Checklist (For Reviewer)
